proxy/cache_test.go

```diff
@@ -262,6 +262,74 @@ func TestCacheExpiration(t *testing.T) {
 	}
 }
 
+func TestCacheExpirationWithTTLOverride(t *testing.T) {
+	dnsProxy := createTestProxy(t, nil)
+	dnsProxy.CacheEnabled = true
+	dnsProxy.CacheMinTTL = 2
+	dnsProxy.CacheMaxTTL = 4
+	err := dnsProxy.Start()
+	if err != nil {
+		t.Fatalf("cannot start the DNS proxy: %s", err)
+	}
+
+	// Create dns messages with 1 second TTL
+	googleReply := dns.Msg{}
+	googleReply.SetQuestion("google.com.", dns.TypeA)
+	googleReply.Response = true
+	googleReply.Answer = []dns.RR{newRR("google.com. 1 IN A 8.8.8.8")}
+
+	youtubeReply := dns.Msg{}
+	youtubeReply.SetQuestion("youtube.com.", dns.TypeA)
+	youtubeReply.Response = true
+	youtubeReply.Answer = []dns.RR{newRR("youtube.com 60 IN A 173.194.221.198")}
+
+	dnsProxy.cache.Set(&youtubeReply)
+	dnsProxy.cache.Set(&googleReply)
+
+	// Wait a minimal amount of time, greater than the
+	// original minimum TTL entry.
+	time.Sleep(1100 * time.Millisecond)
+
+	//Update the expected youtube response TTL
+	//to be maxTTL (4) - time waited (1) = 3
+	youtubeReply.Answer[0].Header().Ttl = 3
```

Comment:
Does cache.Get() really use this value?

Reply:
To be explicit, the code that's highlighted sets up the *expected* version of the response that's returned from the cache.

So yes, when cache.Get() returns it's response, it modifies the TTL to indicate that time has passed since the record was stored. In this case, it's been 1 second.